### PR TITLE
Issue 4 - Add API point to product model

### LINKED ISSUE #4: Add API point to product model
The tastypie API should be added to access product model using RESTful API. By now we should not bother about access rights and authentications. The access should be opened for everybody read/write
Allowed method: `POST`, `GET`, `PUT`, `PATCH`, `DELETE`
API point should support filtering by name with `icontatins`

### PR DESCRIPTION
**Pull request by zitoss**
- Add tastypie module in project settings
- Create folder 'store/api' for REST api resources
- Implement ProductResource with required configuration and possible filtering
- Register ProductResource urls for routing

@sshishov please look at the code

Fixes #4

### CODE REVIEW

File: avtozip/avtozip/settings.py [hunk 1]
```diff
@@ -40,6 +40,9 @@
     'django.contrib.messages',
     'django.contrib.staticfiles',
 
+
```
Comment (sshishov) [inline avtozip/avtozip/settings.py]: Remove 1 extra empty line

File: avtozip/avtozip/urls.py [hunk 2]
```diff
@@ -13,9 +13,16 @@
     1. Import the include() function: from django.conf.urls import url, include
     2. Add a URL to urlpatterns:  url(r'^blog/', include('blog.urls'))
 """
-from django.conf.urls import url
+from django.conf.urls import url, include
 from django.contrib import admin
+from tastypie.api import Api
+from store.api.resources import ProductResource
```
Comment (sshishov) [inline avtozip/avtozip/urls.py:8]: Here after rebase from latest master you will have to add additional empty line

File: avtozip/store/api/resources.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+from store.models import Product
+from django.http import Http404
+
+
+class ProductResource(ModelResource):
+    class Meta:
+        queryset = Product.objects.all()
+        allowed_methods = ['post', 'put', 'get', 'patch', 'delete']
+        filtering = {
+            'name': ('exact', 'icontains',),
```
Comment (sshishov) [inline avtozip/store/api/resources.py]: Please make sure that `icontains` cannot work with `name=<something>`.

File: avtozip/store/api/resources.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+        }
+
+    def get_schema(self, request, **kwargs):
+        # Forbidden schema request
+        raise Http404('Forbidden');
```
Comment (sshishov) [inline avtozip/store/api/resources.py]: Add additional new line at the end of file (and configure your IDE to do it automatically)

Comment (sshishov) [inline avtozip/store/api/resources.py]: HTTP404 error means something not found, but in the description you provided that this information is `forbidden`. Please omit the error message by now completely

File: avtozip/store/api/resources.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+class ProductResource(ModelResource):
+    class Meta:
+        queryset = Product.objects.all()
+        allowed_methods = ['post', 'put', 'get', 'patch', 'delete']
+        filtering = {
+            'name': ('exact', 'icontains',),
+        }
+
+    def get_schema(self, request, **kwargs):
+        # Forbidden schema request
```
Comment (sshishov) [inline avtozip/store/api/resources.py]: And change this comment accordingly after the changes

**Comment by sshishov** [comment]
Please rebase from latest master!

File: avtozip/store/api/resources.py (hunk 3 above)
Comment (zitoss) [inline avtozip/store/api/resources.py]: I've checked this before. You'll get exception when `exact` is absent and you request a data with `name=<something>` to API

File: avtozip/store/api/resources.py (hunk 4 above)
Comment (zitoss) [inline avtozip/store/api/resources.py]: Fixed

File: avtozip/store/api/resources.py (hunk 3 above)
Comment (sshishov) [inline avtozip/store/api/resources.py]: Remove contains by now. I will create new task to substitute exact with contains using custom filters

**Pull request merged**